tests/test_models.py

```diff
@@ -1370,6 +1370,18 @@ def setUp(self):
         self.parent = ProjectFactory(creator=self.user)
         self.node = NodeFactory(creator=self.user, parent=self.parent)
 
+    def test_set_privacy_checks_admin_permissions(self):
+        non_contrib = UserFactory()
+        project = ProjectFactory(creator=self.user, is_public=True)
+        with assert_raises(PermissionsError):
+            project.set_privacy('public', Auth(non_contrib))
+
+        project.set_privacy('public', Auth(project.creator))
+        project.save()
+
+        with assert_raises(PermissionsError):
+            project.set_privacy('public', Auth(non_contrib))
```

Comment:
But the project starts as a public project in this case. Clearly they still shouldn't be able to turn a public project public either, but.